### PR TITLE
fix(tier4_autoware_utils): fix build error (caused by `published_time_publisher.hpp`)

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

FIx build error.

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/9c6f7a56-ef19-55a6-90ba-4b3ec1cb12cb?project_id=prd_jt)

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by xmfcx** [comment]
@satoshi-ota I didn't test this on my machine but, how did the original PRs has passed from build tests then?

- https://github.com/autowarefoundation/autoware.universe/actions/runs/8279452851/job/22671594775

Or a recent general build-and-test:
- https://github.com/autowarefoundation/autoware.universe/actions/runs/8311525279

**Comment by xmfcx** [comment]
Also, what is the build error?

File: common/tier4_autoware_utils/include/tier4_autoware_utils/ros/published_time_publisher.hpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ class PublishedTimePublisher
   {
     bool operator()(const rmw_gid_t & lhs, const rmw_gid_t & rhs) const
     {
-      return std::memcmp(lhs.data, rhs.data, RMW_GID_STORAGE_SIZE) < 0;
+      return memcmp(lhs.data, rhs.data, RMW_GID_STORAGE_SIZE) < 0;
```
Comment (xmfcx) [inline common/tier4_autoware_utils/include/tier4_autoware_utils/ros/published_time_publisher.hpp:94]:
```suggestion
      return std::memcmp(lhs.data, rhs.data, RMW_GID_STORAGE_SIZE) < 0;
```

And could you add:
`#include <cstring>` to the headers list?

- https://en.cppreference.com/w/cpp/string/byte/memcmp

**Comment by takayuki5168** [comment]
@xmfcx 
The following is the build error on the Autoware Evaluator which cannot be reproduced in the GitHub's CI or local environments.
If you have time, please look into it.
image

File: common/tier4_autoware_utils/include/tier4_autoware_utils/ros/published_time_publisher.hpp (hunk 1 above)
Comment (xmfcx) [reply common/tier4_autoware_utils/include/tier4_autoware_utils/ros/published_time_publisher.hpp]: fixed with: https://github.com/autowarefoundation/autoware.universe/pull/6641/commits/1ddfa7b955962ac42dccb9cd6ddbe6a263b59dae

**Comment by xmfcx** [comment]
Thanks for sharing, I will also look into catching this kind of issues with CI too.

**Pull request merged**